From ab4ebac65eae22d8aac40547ae5b0f0cbc870825 Mon Sep 17 00:00:00 2001
From: Aisheng.Dong <b29396@freescale.com>
Date: Fri, 30 Jul 2010 20:07:42 +0800
Subject: [PATCH] ENGR00125772 mx50 esdhc: add DDR support

Add DDR support for eMMC 4.4 cards.
Currently the clock is set to 40Mhz to get good compatibility.

Signed-off-by: Aisheng.Dong <b29396@freescale.com>
---
 arch/arm/mach-mx5/mx50_arm2.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx5/mx50_arm2.c b/arch/arm/mach-mx5/mx50_arm2.c
index 717e37a..87cc9ee 100644
--- a/arch/arm/mach-mx5/mx50_arm2.c
+++ b/arch/arm/mach-mx5/mx50_arm2.c
@@ -550,9 +550,11 @@ static struct mxc_mmc_platform_data mmc2_data = {
 static struct mxc_mmc_platform_data mmc3_data = {
 	.ocr_mask = MMC_VDD_27_28 | MMC_VDD_28_29 | MMC_VDD_29_30
 		| MMC_VDD_31_32,
-	.caps = MMC_CAP_4_BIT_DATA | MMC_CAP_8_BIT_DATA,
+	.caps = MMC_CAP_4_BIT_DATA | MMC_CAP_8_BIT_DATA | MMC_CAP_DATA_DDR,
 	.min_clk = 400000,
-	.max_clk = 50000000,
+	.max_clk = 40000000,
+	.dll_override_en = 1,
+	.dll_delay_cells = 0xc,
 	.card_inserted_state = 0,
 	.status = sdhc_get_card_det_status,
 	.wp_status = sdhc_write_protect,
-- 
1.5.4.4

